seqr/utils/file_utils.py

```diff
@@ -64,3 +64,25 @@ def _google_bucket_file_iter(gs_path, byte_range=None, raw_content=False, user=N
             line = line.decode('utf-8')
         yield line
 
+
+def get_vcf_filename(vcf_path):
+    if vcf_path.endswith('.vcf') or vcf_path.endswith('.vcf.gz') or vcf_path.endswith('.vcf.bgz'):
+        return vcf_path
+    process = subprocess.Popen('gsutil ls ' + vcf_path, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+    if process.wait() != 0:
+        return None
+    for line in process.stdout:
+        file = line.decode('utf-8').strip()
+        if file.endswith('.vcf') or file.endswith('.vcf.gz') or file.endswith('.vcf.bgz'):
+            return file
+    return None
+
+
+def get_vcf_samples(vcf_path):
+    vcf_filename = get_vcf_filename(vcf_path)
+    if not vcf_filename:
+        return {}
+    for line in file_iter(vcf_filename, byte_range=(0, 65536)):
+        if line.startswith('#CHROM'):
+            return set(line.rstrip().split('\tFORMAT\t', 2)[1].split('\t') if line.endswith('\n') else [])
```

Comment:
@mike-w-wilson I'm not super familiar with VCF formats - can we safely assume that any VCF header (or at least any VCF that would work with our loading pipeline)) would include  a tab-separated list of sample IDs after the `FORHAT` header?

Reply:
Yes, that is the spec put out by the official docs. All samples are in the header and all entires are tab-separated -- though I guess someone could technically change it to spaces but this does not conform to the VCF spec so I don't think we need to support that.
```
1 The VCF specification
VCF is a text file format (most likely stored in a compressed manner). It contains meta-information lines (prefixed
with “##”), a header line (prefixed with “#”), and data lines each containing information about a position in the
genome and genotype information on samples for each position (text fields separated by tabs). Zero length fields are
not allowed, a dot (“.”) must be used instead. In order to ensure interoperability across platforms, VCF compliant
implementations must support both LF (“\n”) and CR+LF (“\r\n”) newline conventions.
...
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">
##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality">
#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.
```

Reply:
I agree we do not have to support VCFs that do not conform to spec